tests/Integration/Schema/Directives/WithCountDirectiveTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Tests\Integration\Schema\Directives;
+
+use Nuwave\Lighthouse\Support\AppVersion;
+use Tests\DBTestCase;
+use Tests\Utils\Models\Task;
+use Tests\Utils\Models\User;
+
+class WithCountDirectiveTest extends DBTestCase
+{
+    public function testEagerLoadsRelationCount(): void
+    {
+        if (AppVersion::below(5.7)) {
+            $this->markTestSkipped('Version less than 5.7 do not support loadCount().');
+        }
+
+        $this->schema = /** @lang GraphQL */ '
+        type Query {
+            user: User @first
+        }
+
+        type User {
+            tasks_count: Int! @withCount
+            tasks_relation_count: Int! @withCount(relation: "tasks")
```

Comment:
I removed the `$relation as $field` part of the directive, so this fails for now. The semantics of this are a bit unclear to me. It is not described as part of the `@withCount` directive definition.

For example, what would happen in a case like this, assuming the user has a column `name`:

```graphql
type User {
  name: Int! @withCount(relation: "tasks")
}
```

Would the eager-load supersede the column?

Reply:
> For example, what would happen in a case like this, assuming the user has a column name:
> ```graphql
> type User {
>   name: Int! @withCount(relation: "tasks")
> }
>```
>Would the eager-load supersede the column?

Yes, I think it should supersede it since the field may or may not follow the *_count* conventions. For example, `completed_tasks` could return how many tasks are completed on a task list. It uses the underlying `tasks` relation but should be aliased to `completed_tasks` when fetched from the database and returned as so allowing it to be resolved properly. This follows Laravel's `withCount()` aliasing capabilities:

https://github.com/laravel/framework/blob/7d30c1b564ddcd87a0fb6b59bc7b8d3cde7432e8/src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php#L368-L378

>Ultimately, i think we are better off by not conflating the resolving of the field with eager >loading. They might be completely unrelated, e.g.:
> ```graphql
> type User {
>   statistics: UserStatistics @method @withCount(relation: "tasks")
> }
> ```

In this case, the `statistics()` method on `User::class` would use `$rootValue->tasks_count` to resolve the `UserStatistics`. I don't see an issue with that.